src/qiskit_qec/utils/stim_tools.py

```diff
@@ -0,0 +1,347 @@
+            return
+        if len(dets) == 1:
+            dets = [dets[0], model.num_detectors]
+            # if frame_changes == []:
+            #     dets = [dets[0], model.num_detectors]
+            # else:
+            #     dets = [dets[0], model.num_detectors+1]
+        if len(dets) > 2:
+            raise NotImplementedError(
+                f"Error with more than 2 symptoms can't become an edge or boundary edge: {dets!r}."
+            )
+        if g.has_edge(dets[0], dets[1]):
+            edge_ind = list(g.edge_list()).index((dets[0], dets[1]))
+            edge_data = g.edges()[edge_ind].properties
+            old_p = edge_data["error_probability"]
+            old_frame_changes = edge_data["fault_ids"]
+            # If frame changes differ, the code has distance 2; just keep whichever was first.
+            if set(old_frame_changes) == set(frame_changes):
+                p = p * (1 - old_p) + old_p * (1 - p)
+                g.remove_edge(dets[0], dets[1])
+        if p > 0.5:
+            p = 1 - p
+        if p > 0:
+            qubits = list(
+                set(index_to_DecodingGraphNode[dets[0]].qubits).intersection(
+                    index_to_DecodingGraphNode[dets[1]].qubits
+                )
+            )
+            edge = DecodingGraphEdge(
+                qubits=qubits,
+                weight=log((1 - p) / p),
+                properties={"fault_ids": set(frame_changes), "error_probability": p},
+            )
+            g.add_edge(dets[0], dets[1], edge)
+            hyperedge[dets[0], dets[1]] = edge
+
+    hyperedges = []
+
+    for instruction in model:
+        if isinstance(instruction, StimDemInstruction):
+            if instruction.type == "error":
+                dets: List[int] = []
+                frames: List[int] = []
+                t: StimDemTarget
+                p = instruction.args_copy()[0]
+                hyperedge = {}
+                for t in instruction.targets_copy():
+                    if t.is_relative_detector_id():
+                        dets.append(t.val)
+                    elif t.is_logical_observable_id():
+                        frames.append(t.val)
+                    elif t.is_separator():
+                        # Treat each component of a decomposed error as an independent error.
+                        handle_error(p, dets, frames, hyperedge)
+                        frames = []
+                        dets = []
+                # Handle last component.
+                handle_error(p, dets, frames, hyperedge)
+                if len(hyperedge) > 1:
+                    hyperedges.append(hyperedge)
+            elif instruction.type == "detector":
+                pass
+            elif instruction.type == "logical_observable":
+                pass
+            else:
+                raise NotImplementedError()
+        else:
+            raise NotImplementedError()
+
+    return g, hyperedges
+
+
+def noisify_circuit(circuits, noise_model):
```

Comment:
Can you give more information in the comments/typing for what noise_model should be? For example specify if it's a dictionary with typing, and give more details in the Args what the keys/values are. 
With circuits arg, can you specify with typing that it could be a qiskit quantum circuits or list of quantum circuits